base_layer/p2p/src/executor/handles/and_then_service.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019 The Tari Project
```

Comment:
You should submit this as a PR to the Tower repo, if it it's generally useful

Reply:
Was thinking that it could potentially be a `FutureExt` combinator for tower services, will look into it